web/src/main/java/org/springframework/security/web/server/authentication/session/ServerSessionAuthenticationStrategy.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.server.authentication.session;
+
+import reactor.core.publisher.Mono;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.server.WebFilterExchange;
+import org.springframework.security.web.server.authentication.AuthenticationWebFilter;
+
+/**
+ * Allows pluggable support for WebSession-related behaviour when an authentication
+ * occurs.
+ *
+ * @author Marcus da Coregio
+ * @since 6.3
+ * @see AuthenticationWebFilter#setSessionAuthenticationStrategy(ServerSessionAuthenticationStrategy)
+ */
+public interface ServerSessionAuthenticationStrategy {
```

Comment:
I wonder if a new interface is necessary in this case since we already have `ServerAuthenticationSuccessHandler`, whose method signature is the same.

It seems like the implementations could be added to a `DelegatingServerAuthenticationSuccessHandler` instance. Do you try that already?

Reply:
Thanks! I've changed the implementation to use the `ServerAuthenticationSuccessHandler` instead of a new interface.